TaskManagement/src/main/java/com/taskManagement/controllers/TasksController.java

```diff
@@ -23,120 +24,93 @@
+import com.taskManagement.util.CollectionUtil;
 import com.taskManagement.validators.TaskValidator;
 
 @Controller
 @RestController
-@RequestMapping(path="/tasks") // This means URL's will start with /tasks after application path
+@RequestMapping(path = "/tasks") // This means URL's will start with /tasks after application path
 public class TasksController {
+	private TaskValidator validator = new TaskValidator();
 
 	@Autowired
-	private TaskRepository taskRepository;
-	private TaskValidator validator = new TaskValidator();
-	
-	//POST :  Create task
-	@RequestMapping(method=RequestMethod.POST)
-	public ResponseEntity<?> addNewTask(@RequestBody Task task) {
-		if(validator.isValidTask(task) == false) {
-			return new ResponseEntity<String>("In valid task",HttpStatus.BAD_REQUEST);
-		}
-		taskRepository.save(task);
-		return new ResponseEntity<Task>(task,HttpStatus.CREATED);
-	}
-	
-	//PUT : Update a task
-	@RequestMapping(method=RequestMethod.PUT,path = "/{id}")
-	public ResponseEntity<?> updateTask(@PathVariable("id") long id, @RequestBody Task task){
-		Optional<Task> optionalTask = taskRepository.findById(id);
-		if(optionalTask.isPresent() == false) {
-			return new ResponseEntity<String>("Not found",HttpStatus.NOT_FOUND);
-		}
-		
-		if(validator.isValidTask(task) == false) {
-			return new ResponseEntity<String>("In valid task",HttpStatus.BAD_REQUEST);
+	private TaskService taskService;
+
+	// POST : Create task
+	@RequestMapping(method = RequestMethod.POST)
+	public ResponseEntity<?> add(@RequestBody Task task) {
+		if (validator.isValidTask(task) == false) {
+			return new ResponseEntity<String>("In valid task", HttpStatus.BAD_REQUEST);
 		}
 		
-		Task currentTask = optionalTask.get();
-		currentTask.setState(task.getState());
-		currentTask.setDescription(task.getDescription());
-		
-		taskRepository.save(currentTask);
-		return new ResponseEntity<Task>(currentTask,HttpStatus.ACCEPTED);
+		taskService.save(task);
+		return new ResponseEntity<Task>(task, HttpStatus.CREATED);
 	}
-	
-	//PATH : Update state of task
-	@RequestMapping(method=RequestMethod.PATCH,path = "/{id}")
-	public ResponseEntity<?> updateStateOfTask(@PathVariable("id") long id,@RequestBody Task task){
-		Optional<Task> optionalTask = taskRepository.findById(id);
-		if(optionalTask.isPresent() == false) {
-			return new ResponseEntity<String>("Not found",HttpStatus.NOT_FOUND);
+
+	// PUT : Update a task
+	@RequestMapping(method = RequestMethod.PUT, path = "/{id}")
+	public ResponseEntity<?> update(@PathVariable("id") long id, @RequestBody Task task) {
+		if (validator.isValidTask(task) == false) {
+			return new ResponseEntity<String>("In valid task", HttpStatus.BAD_REQUEST);
 		}
-		
-		Task currentTask = optionalTask.get();
-		if(null != task.getState()) {
-			currentTask.setState(task.getState());
+
+		try {
+			taskService.update(id, task);
+			return new ResponseEntity<Task>(task, HttpStatus.ACCEPTED);
+		} catch (TaskExceptions e) {
```

Comment:
TaskException can be anything. For this, you can not return a NOT_FOUND error code.